### DISCUSSION TITLE
Should filesystems.cloud be deprecated?

### DISCUSSION THREAD

**Discussion by chrismcintosh** [laravel/framework#49750] [Q&A]
I was recently configuring file storage for an application and a specific need lead me to discover the filesystems.cloud config option.

Relevant Code:
[https://github.com/search?q=repo%3Alaravel%2Fframework%20filesystems.cloud&type=code](https://github.com/search?q=repo%3Alaravel%2Fframework%20filesystems.cloud&type=code)

Despite my attempts to utilize this option, it seems to have no effect. 

There isn't a stub in the filesystem config for where I would imagine this would get configured
https://github.com/laravel/laravel/blob/10.x/config/filesystems.php#L17

Additionally there doesn't appear to have any documentation for this feature when searching all variations of
- FILESYSTEM_CLOUD
- filesystems.cloud

I'm curious whether this code is intended for legacy support, if there's a crucial aspect of this feature that I might be misunderstanding, or if it's possibly a candidate for deprecation. 

Maybe I'm not understanding something, but for me the presence of this code lead to confusion.

Thanks for any help or direction y'all are able to provide!

**Comment by sh4msi** [comment]
To use ```FILE SYSTEM CLOUD```, you must first configure it

For example:
```
        'minio' => [
            'driver' => 's3',
            'key' => env('MINIO_ACCESS_KEY'),
            'secret' => env('MINIO_SECRET_KEY'),
            'region' => env('MINIO_DEFAULT_REGION'),
            'bucket' => env('MINIO_BUCKET'),
            'url' => env('MINIO_URL'),
            'endpoint' => env('MINIO_ENDPOINT'),
            'use_path_style_endpoint' => env('MINIO_USE_PATH_STYLE_ENDPOINT', true),
        ],
```
And then :
```
'cloud' => env('FILESYSTEM_CLOUD', 'minio'),
```
Now you can use it with the following command:
```
Storage::cloud()->url('name')
```
